segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/NoOpChunkStorageProvider.java

```diff
@@ -0,0 +1,190 @@
+                    fromOffset, bufferOffset, length, buffer.length));
+        }
+
+        return length;
+    }
+
+    @Override
+    protected int doWrite(ChunkHandle handle, long offset, int length, InputStream data) throws ChunkStorageException, IndexOutOfBoundsException {
+
+        ChunkData chunkData = chunkMetadata.get(handle.getChunkName());
+        if (null == chunkData) {
+            throw new ChunkNotFoundException(handle.getChunkName());
+        }
+        if (offset != chunkData.length) {
+            throw new IndexOutOfBoundsException("");
+        }
+        if (chunkData.isReadonly) {
+            throw new ChunkStorageException(handle.getChunkName(), "chunk is readonly");
+        }
+        chunkData.length = offset + length;
+        chunkMetadata.put(handle.getChunkName(), chunkData);
+        return length;
+    }
+
+    @Override
+    protected int doConcat(ConcatArgument[] chunks) throws ChunkStorageException, UnsupportedOperationException {
+        int total = 0;
+        for (ConcatArgument chunk : chunks) {
+            val chunkData = chunkMetadata.get(chunk.getName());
+            Preconditions.checkState(null != chunkData);
+            Preconditions.checkState(chunkData.length >= chunk.getLength());
+            total += chunk.getLength();
+        }
+
+        val targetChunkData = chunkMetadata.get(chunks[0].getName());
+        targetChunkData.length = total;
+        for (int i = 1; i < chunks.length; i++) {
+            chunkMetadata.remove(chunks[i].getName());
+        }
+
+        return total;
+    }
+
+    @Override
+    protected boolean doTruncate(ChunkHandle handle, long offset) throws ChunkStorageException, UnsupportedOperationException {
+        ChunkData chunkData = chunkMetadata.get(handle.getChunkName());
+        if (null == chunkData) {
+            throw new ChunkNotFoundException(handle.getChunkName());
+        }
+        if (offset < chunkData.length) {
+            chunkData.length = offset;
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    protected boolean doSetReadOnly(ChunkHandle handle, boolean isReadOnly) throws ChunkStorageException, UnsupportedOperationException {
+        Preconditions.checkNotNull(null != handle, "handle");
+        Preconditions.checkNotNull(handle.getChunkName(), "handle");
+        String chunkName = handle.getChunkName();
+        ChunkData chunkData = chunkMetadata.get(chunkName);
+        if (null == chunkData) {
+            throw new ChunkNotFoundException(chunkName);
+        }
+        chunkData.isReadonly = isReadOnly;
+        return false;
+    }
+
+    @Override
+    public void addChunk(String chunkName, long length) {
+        ChunkData chunkData = new ChunkData();
+        chunkData.length = length;
+        chunkMetadata.put(chunkName, chunkData);
+    }
+
+    /**
+     * Stores the chunk data.
+     */
+    public static class ChunkData {
+        public long length;
```

Comment:
final?

Reply:
This field gets modified every time we write.